vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/ReplaceSourceMapInDocExpression.java

```diff
@@ -102,6 +102,14 @@ public boolean rewritePost(Mutable<ILogicalOperator> opRef, IOptimizationContext
                     modified = true;
                 }
             }
+            docExpression = ExpressionToolbox.findFirstFunctionExpression(expression,
+                    BuiltinFunctions.FN_DOC_JSON_1.getFunctionIdentifier());
```

Comment:
Should be called jn:json-doc [1]

[1] http://jsoniq.org/docs/JSONiqExtensionToXQuery/html-single/index.html#idm139680622843744